packages/dataviews/src/components/dataviews-view-config/index.tsx

```diff
@@ -226,45 +227,174 @@ function ItemsPerPageControl() {
 	);
 }
 
+function isFieldHidable(
+	field: NormalizedField< any >,
+	mandatoryFields: string | any[]
+) {
+	return (
+		field.enableHiding !== false && ! mandatoryFields.includes( field.id )
+	);
+}
+
 function FieldControl() {
 	const { view, fields, onChangeView } = useContext( DataViewsContext );
 	const mandatoryFields = getMandatoryFields( view );
-	const hidableFields = fields.filter(
-		( field ) =>
-			field.enableHiding !== false &&
-			! mandatoryFields.includes( field.id )
-	);
 	const viewFields = view.fields || fields.map( ( field ) => field.id );
-	if ( ! hidableFields?.length ) {
+	const fieldsOrdered = useMemo( () => {
+		if ( ! view.fields ) {
+			return fields.map( ( field ) => field.id );
+		}
+		const result = [ ...view.fields ];
+		fields.forEach( ( field ) => {
+			if ( result.includes( field.id ) ) {
+				return;
+			}
+			result.push( field.id );
+		} );
+		return result;
+	}, [ fields, view.fields ] );
+	if (
+		view.type !== LAYOUT_TABLE &&
+		! fields?.some( ( field ) => isFieldHidable( field, mandatoryFields ) )
+	) {
 		return null;
 	}
 	return (
 		<ItemGroup isBordered isSeparated>
-			{ hidableFields?.map( ( field ) => {
+			{ fieldsOrdered?.map( ( fieldId ) => {
+				const field = fields.find(
+					( f ) => f.id === fieldId
+				) as NormalizedField< any >;
+				const index = view.fields?.indexOf( field.id ) as number;
+				const isHidable = isFieldHidable( field, mandatoryFields );
+				if ( view.type !== LAYOUT_TABLE && ! isHidable ) {
+					return null;
+				}
 				const isVisible = viewFields.includes( field.id );
 				return (
 					<Item key={ field.id }>
-						<HStack expanded>
+						<HStack
+							expanded
+							className="dataviews-field-control__field"
+						>
 							<span>{ field.label }</span>
-							<Button
-								size="compact"
-								onClick={ () =>
-									onChangeView( {
-										...view,
-										fields: isVisible
-											? viewFields.filter(
-													( id ) => id !== field.id
+							<HStack
+								justify="flex-end"
+								expanded={ false }
+								className="dataviews-field-control__actions"
+							>
+								{ view.type === LAYOUT_TABLE && (
+									<>
+										<Button
+											disabled={
+												! isVisible || index < 1
+											}
+											accessibleWhenDisabled={ false }
```

Comment:
Is there any reason not to use `accessibleWhenDisabled={ true }` here? Ideally the buttons are still accessible even when disabled.